endpoints/data_objects/src/main.cpp

```diff
@@ -610,12 +961,27 @@ namespace
 				}
 				auto remaining_bytes = std::stoll(iter->second);
 
+				if (remaining_bytes <= 0) {
+					log::error("{}: Invalid value for [count] parameter. Expected an integer greater than 0.", fn);
+					return _sess_ptr->send(irods::http::fail(
+						res, http::status::bad_request)); // TODO Return a JSON object with these error messages.
+				}
+
 				iter = _args.find("bytes");
 				if (iter == std::end(_args)) {
 					log::error("{}: Missing [bytes] parameter.", fn);
 					return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
 				}
 
+				// TODO Should we require the count and total bytes to be equal?
+				if (std::cmp_greater(remaining_bytes, iter->second.size())) {
+					log::error(
+						"{}: The number of bytes to write exceeds the number of bytes available: [count] > number of "
+					    "[bytes]",
+						fn);
+					return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+				}
```

Comment:
Handle TODO.

Reply:
yes... unless perhaps count isn't present?

Reply:
As currently implemented, clients are allowed to pass MORE bytes than the value specified in `count`. But I think forcing devs to pass values that align is better in the long run.

Will update to require the number of bytes and the count to be identical.

Reply:
agreed they should match if they're both present.

but we'd also talked about them not having to provide the count at all...

Reply:
True, but I'm a little hesitant to remove it _(I don't have any real reason behind that either)_. I'm pretty sure it's safe to remove altogether because the payload is known upfront. Let me double check and if there's no reason to keep it, I'll remove it.

Reply:
right - if we have the actual bytes, we should be able to determine the count.  and if the count *is* present and incorrect... that feels like an error case.

almost... feels like count might be unnecessary then?  hm...   what does having it protect us from?

Reply:
> what does having it protect us from?

That's what I need to investigate. Proper HTTP requests do not need to have a count provided unless it's an optimization thing (and that's going to be based on the API's design).

Reply:
gotcha.

Reply:
Removed the `count` param from `op_write` and all tests passed.

Reply:
Forgot to mention that `count` is essentially initialized to the size of the incoming data to write.

Reply:
perfect.   so `count` is no longer a thing that a client even sees / needs to send?   ever?